Clarify build instructions in mesos/docker guide

karlkfi opened the pull request:
- Linux build requires docker sudo group
- Golang & Make optional

xref: https://github.com/mesosphere/kubernetes-mesos/issues/305

Commit:
Clarify build instructions in mesos/docker guide

- Linux build requires docker sudo group
- Golang & Make optional
- Extracted a section on building locally

air commented:
Yep let's try this and see how it flows.

sttts commented on docs/getting-started-guides/mesos-docker.md:
````diff
@@ -154,22 +161,38 @@ host machine (mac).
+    ```
 
-    Unless you're on linux, you'll also need to build the binaries for linux/amd64 (for the docker containers):
+    If you don't have or want Golang installed, you can build Kubernetes for linux/amd64 in a docker container:
````
Do we really need this many cases?

Why not just
```
KUBERNETES_CONTRIB=mesos
make
build/run.sh hack/build-go.sh (optional for linux)
```
Full stop. For more information /docs/development.md.

karlkfi replied:
The primary reason is because I think people new to k8sm should have to install as few things as possible. Non-developers should be able to deploy k8sm without installing a programming language. CI's also should install as little as possible, and this documents the alternatives.

sttts replied:
My only fear is that we document more than we can maintain. Then we have verbose documentation which is outdated and wrong.

karlkfi commented:
I simplified the guide step to the simplest thing that works and extracted a section on building locally for developers to do it faster, if they have Go and Make installed.

sttts commented:
Only a typo. Otherwise, lgtm :+1:

karlkfi commented:
More of a baseball analogy than a typo... How much do you care? ;)

sttts commented:
Well, bases is fine if it sounds good in your native ears.

karlkfi commented:
http://idioms.thefreedictionary.com/cover+all+the+bases

karlkfi commented:
rebased & squashed

karlkfi commented:
bases -> cases

sttts commented:
Lgtm. Didn't test the given command lines myself though.
